### DISCUSSION TITLE
How to query for notifications with specific params

### DISCUSSION THREAD

**Discussion by joemasilotti** [excid3/noticed#220] [General]
```ruby
class Conversation < ApplicationRecord
  has_many :messages
end

class Message < ApplicationRecord
  belongs_to :conversation
  belongs_to :user

  has_noticied_notifications

  after_create_commit :send_recipient_notification

  def send_recipient_notification
    NewMessageNotification.with(message: self, conversation:).deliver_later(user)
  end
end
```

I'm trying to query for the number of "read" conversations. Or, how many conversations have one or more messages with a read notification. I tried something like this but can't get the query to work with the `params` jsonb column from Noticed.

```ruby
Notification.read.where(type: "NewMessageNotification").group(params: {conversation: :id}).count
```

Am I thinking about this the wrong way? Or do I need to brush up on my jsonb/Arel querying?

**Comment by excid3** [comment]
Depends on what you're trying to do. You can query against the `read_at` column or if you need params, you can query against that. 

Postgres uses the `->` syntax to query jsonb columns or you can use `@>` to look for an object. https://github.com/excid3/noticed/blob/master/lib/noticed/has_notifications.rb#L18-L31
